File: bungee/src/main/java/live/turna/phenyl/bungee/listener/MiraiListenerManager.java
package live.turna.phenyl.bungee.listener;

import live.turna.phenyl.bungee.event.BungeeBotOfflineEvent;
import live.turna.phenyl.bungee.event.BungeeGroupMessageEvent;
import live.turna.phenyl.common.listener.AbstractMiraiListenerManager;
import net.mamoe.mirai.Bot;
import net.mamoe.mirai.event.EventChannel;
import net.mamoe.mirai.event.events.BotEvent;
import net.mamoe.mirai.event.events.BotOfflineEvent;
import net.mamoe.mirai.event.events.GroupMessageEvent;
import net.md_5.bungee.api.ProxyServer;

/**
 * <b>MiraiListenerManager</b><br>
 * *
 *
 * @since 2022/5/6 15:42
 */
public class MiraiListenerManager extends AbstractMiraiListenerManager {
    @Override
    public void start(Bot bot) {
        EventChannel<BotEvent> eventChannel = bot.getEventChannel();

        BotOfflineListener = eventChannel.subscribeAlways(BotOfflineEvent.class, e -> ProxyServer.getInstance().getPluginManager().callEvent(new BungeeBotOfflineEvent(e)));
        GroupMessageListener = eventChannel.subscribeAlways(GroupMessageEvent.class, e -> ProxyServer.getInstance().getPluginManager().callEvent(new BungeeGroupMessageEvent(e)));
    }
}

File: bungee/src/main/java/live/turna/phenyl/bungee/listener/MiraiListenerManager.kt
package live.turna.phenyl.bungee.listener

import live.turna.phenyl.bungee.event.BungeeBotOfflineEvent
import live.turna.phenyl.bungee.event.BungeeGroupMessageEvent
import live.turna.phenyl.common.listener.AbstractMiraiListenerManager
import net.mamoe.mirai.Bot
import net.mamoe.mirai.event.events.BotOfflineEvent
import net.mamoe.mirai.event.events.GroupMessageEvent
import net.md_5.bungee.api.ProxyServer

class MiraiListenerManager : AbstractMiraiListenerManager() {
    override fun start(bot: Bot) {
        val eventChannel = bot.eventChannel
        BotOfflineListener =
            bot.eventChannel.subscribeAlways<BotOfflineEvent> { e ->
                ProxyServer.getInstance().pluginManager.callEvent(
                    BungeeBotOfflineEvent(e)
                )
            }
        GroupMessageListener = eventChannel.subscribeAlways<GroupMessageEvent> { e ->
            ProxyServer.getInstance().pluginManager.callEvent(
                BungeeGroupMessageEvent(e)
            )
        }
    }
}